config/runtimes/vllm-multinode-template.yaml

```diff
@@ -0,0 +1,259 @@
+
+                  # Determine health status based on GPU usage
+                  if [[ "$used_gpu" != "$reserved_gpu" ]]; then
+                    echo "$1: Unhealthy - Used: $used_gpu, Reserved: $reserved_gpu"
+                    exit 1
+                  fi
+          readinessProbe:
+            failureThreshold: 2
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 5
+            exec:
+              command:
+                - bash
+                - -c
+                - |
+                  # Check if the registered nodes count matches PIPELINE_PARALLEL_SIZE
+                  registered_node_count=$(ray status | grep -c node_)
+                  if [[ $registered_node_count -ne "$PIPELINE_PARALLEL_SIZE" ]]; then
+                    echo "Readiness Probe: Unhealthy - Registered nodes count ($registered_node_count) does not match PIPELINE_PARALLEL_SIZE ($PIPELINE_PARALLEL_SIZE)."
+                    exit 1
+                  fi
+
+                  # Check if the registered ray nodes count is the same as PIPELINE_PARALLEL_SIZE
+                  gpu_status=$(ray status | grep GPU)
+                  if [[ -z $gpu_status ]]; then
+                    echo "$1: GPU does not exist"
+                    exit 1
+                  fi
+
+                  used_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f1)
+                  reserved_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f2)
+
+                  # Determine health status based on GPU usage
+                  if [[ "$used_gpu" != "$reserved_gpu" ]]; then
+                    echo "$1: Unhealthy - Used: $used_gpu, Reserved: $reserved_gpu"
+                    exit 1
+                  fi
+                  
+                  # Check model health
+                  curl --silent --max-time 5 --fail-with-body http://localhost:8080/health
+          startupProbe:
+            failureThreshold: 40
+            periodSeconds: 30
+            successThreshold: 1
+            timeoutSeconds: 5
+            exec:
+              command:
+                - bash
+                - -c
+                - |
+                  ray status
+
+                  # Check model health
+                  curl --silent --max-time 5 --fail-with-body http://localhost:8080/health
+          ports:
+            - containerPort: 8080
+              name: http
+              protocol: TCP
+      volumes:
+        - name: shm
+          emptyDir:
+            medium: Memory
+            sizeLimit: 12Gi    
+      workerSpec:
+        pipelineParallelSize: 2
+        tensorParallelSize: 1
+        containers:
+          - name: worker-container
+            image: $(vllm-image)
+            command: ["bash", "-c"]
+            args:
+              - |
+                SECONDS=0
+
+                while true; do              
+                  if (( SECONDS <= 120 )); then
```

Comment:
The `SECONDS` variable in Bash is a special variable that automatically increments, representing the number of seconds that have elapsed since the script started running. Therefore, it’s not necessary to manually increment SECONDS in each iteration—it will automatically accumulate the elapsed time.